python/tank/template.py

```diff
@@ -439,12 +439,71 @@ def parent(self):
         """
         parent_definition = os.path.dirname(self.definition)
         if parent_definition:
-            return TemplatePath(parent_definition, self.keys, self.root_path, None)
+            return TemplatePath(parent_definition, self.keys, self.root_path, None, self._root_paths_all_os)
         return None
 
-    def _apply_fields(self, fields, ignore_types=None):
-        relative_path = super(TemplatePath, self)._apply_fields(fields, ignore_types)
-        return os.path.join(self.root_path, relative_path) if relative_path else self.root_path
+    def _apply_fields(self, fields, ignore_types=None, platform=None):
+        """
+        Creates path using fields.
+
+        :param fields: Mapping of keys to fields. Keys must match those in template 
+                       definition.
+        :param ignore_type: Keys for whom the defined type is ignored as list of strings.
+                            This allows setting a Key whose type is int with a string value.
+        :param platform: Optional operating system platform. If you leave it at the 
+                         default value of None, paths will be created to match the 
+                         current operating system. If you pass in a platform constant
+                         (e.g. Template.PLATFORM_XXX), paths will be generating to 
+                         target that specific platform.
+
+        :returns: Full path, matching the template with the given fields inserted.
+        """        
+        relative_path = super(TemplatePath, self)._apply_fields(fields, ignore_types, platform)
+        
+        if platform is None:
+            # return the current OS platform's path
+            return os.path.join(self.root_path, relative_path) if relative_path else self.root_path
+    
+        else:
+            # caller has requested a path for another OS
+            
+            if self._root_paths_all_os is None:
+                # it's possible that the additional os paths are not set for a template
+                # object (mainly because of backwards compatibility reasons) and in this case
+                # we cannot compute the path.
+                raise TankError("Template %s cannot resolve path for operating system '%s' - "
+                                "it was instantiated in a mode which only supports the resolving "
+                                "of current operating system paths." % (self, platform))
+            
+            PATH_LOOKUP = {self.PLATFORM_LINUX: "linux2", 
```

Comment:
This feels like we are mapping one constant to another - couldn't PLATFORM_XXX just be defined as their respective strings instead of range(3)?  Then this code wouldn't be needed.